### PR TITLE
Unit test with Julienne

### PR DESCRIPTION
**Pull request by rouson**
Because the Veggies unit testing framework exposes what appears to be an internal compiler error in LLVM Flang (`flang-new`), this PR refactors the Caffeine test suite to replace Veggies with [Julienne], which provides similar functionality and supports Flang. This commit also removes the unused iso_varying_string dependency and renames all `test/caf_*` files to `test/prif_*`.

To Do
------

- [x] Update `fpm.toml.template`: replace veggies with julienne and remove the unused iso_varying_string dependency
- [x] Replace main test program
- [x] Refactor/rename prif_init test module
- [x] Refactor/rename prif_allocate test module
- [x] Refactor/rename prif_co_broadcast test module
- [x] Refactor/rename prif_co_max test module
- [x] Refactor/rename prif_co_min test module
- [x] Refactor/rename prif_co_reduce test module
- [x] Refactor/rename prif_co_sum test module
- [x] Refactor/rename prif_error_stop test module
- [x] Refactor/rename prif_image_index test module
- [x] Refactor/rename prif_num_images test module
- [x] Refactor/rename prif_rma test module
- [x] Refactor/rename prif_stop test module
- [x] Refactor/rename prif_teams test module
- [x] Refactor/rename prif_this_image test module

[Julienne]: https://github.com/berkeleylab/julienne

### CODE REVIEW

**Review by bonachea — CHANGES_REQUESTED** [review]
I realize this is still Draft, but I provided a few suggestions on preprocessor style.

**Review by bonachea — CHANGES_REQUESTED** [review]

File: test/main.F90 [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+#else
+    call prif_co_max_test%report(passes, tests)
+    call prif_co_min_test%report(passes, tests)
+    call prif_co_reduce_test%report(passes, tests)
+    call prif_co_sum_test%report(passes, tests)
+    call prif_error_stop_test%report(passes, tests)
+    call prif_image_index_test%report(passes, tests)
+    call prif_rma_test%report(passes, tests)
+    call prif_stop_test%report(passes, tests)
+#endif
```
Comment (everythingfunctional) [inline test/main.F90:99]: This evokes a somewhat philosophical question about how to proceed in the face of known failures.

Comment (rouson) [reply test/main.F90:99]: @everythingfunctional that's a solid point.  At this point, I'd like to understand and address the issues even if the only outcome is that Caffeine and/or Julienne improve whether or not we decide for Caffeine to use Julienne.  So far, these issues have helped to identify problems in both packages that I'm working on addressing.  Most of the above failures relate the handling of `character` stop codes in error termination (see issue #77).  I'm making good progress with addressing the above failures and hope to have most or all fixed by the time of tomorrow's call.

File: src/caffeine/unit_test_parameters_m.f90
Comment (everythingfunctional) [inline src/caffeine/unit_test_parameters_m.f90:1]: This feels like it doesn't belong in the main library. I'd suggest instead making another fpm project in this repo, i.e. `fpm new --lib caffeine-test-helpers`, putting this in there, and then specifying that just the tests and those examples depend on it. I.e.

```
[dev-dependencies]
caffeine-test-helpers = { path = "./caffeine-test-helpers" }
```

Comment (rouson) [reply src/caffeine/unit_test_parameters_m.f90:1]: @everythingfunctional good suggestion.  I either missed or forgot this comment.  I'm creating a new test-with-julienne branch and corresponding pull request to replace the previous switch-to-julienne branch and this pull request.   On the new branch, this file is removed.

**Comment by rouson** [comment]
PR #169 supersedes this PR.

**Pull request closed without merging**